exo-go/src/app_runner/index.go

```diff
@@ -0,0 +1,100 @@
+package appRunner
+
+import (
+	"fmt"
+	"math"
+	"path"
+	"sync"
+
+	"github.com/Originate/exosphere/exo-go/src/app_config_helpers"
+	"github.com/Originate/exosphere/exo-go/src/docker_compose"
+	"github.com/Originate/exosphere/exo-go/src/logger"
+	"github.com/Originate/exosphere/exo-go/src/service_config_helpers"
+	"github.com/Originate/exosphere/exo-go/src/types"
+	"github.com/fatih/color"
+)
+
+// AppRunner runs the overall application
+type AppRunner struct {
+	AppConfig            types.AppConfig
+	Logger               *logger.Logger
+	Env                  map[string]string
+	DockerConfigLocation string
+	OnlineTexts          map[string]string
+}
+
+// NewAppRunner is AppRunner's constructor
+func NewAppRunner(appConfig types.AppConfig, logger *logger.Logger, cwd string) *AppRunner {
+	appRunner := &AppRunner{AppConfig: appConfig, Logger: logger, Env: appConfigHelpers.GetEnvironmentVariables(appConfig), DockerConfigLocation: path.Join(cwd, "tmp")}
+	return appRunner
+}
+
+func (appRunner *AppRunner) compileOnlineTexts() (map[string]string, error) {
+	onlineTexts := make(map[string]string)
+	for _, dependency := range appRunner.AppConfig.Dependencies {
+		onlineTexts[dependency.Name] = dependency.GetOnlineText()
+	}
+	serviceConfigs, err := serviceConfigHelpers.GetServiceConfigs(appRunner.AppConfig)
+	if err != nil {
+		return map[string]string{}, err
+	}
+	for serviceName, serviceConfig := range serviceConfigs {
+		onlineTexts[serviceName] = serviceConfig.Startup["online-text"]
+	}
+	return onlineTexts, nil
+}
+
+func (appRunner *AppRunner) getEnv() []string {
+	formattedEnvVars := []string{}
+	for variable, value := range appRunner.Env {
+		formattedEnvVars = append(formattedEnvVars, fmt.Sprintf("%s=%s", variable, value))
+	}
+	return formattedEnvVars
+}
+
+// Shutdown shuts down the application and returns the process output and an error if any
+func (appRunner *AppRunner) Shutdown(closeMessage, errorMessage string) (string, error) {
```

Comment:
Thoughts on making the input to this be a struct (maybe named `ShutdownConfig`) which has two properties closeMessage and errorMessage. The current interface is a bit awkward to me as you always need to specify an empty errorMessage or closeMessage which wouldn't be necessary with an object since it would default to an empty string